miden-tx/src/testing/tx_context/mod.rs

```diff
@@ -94,7 +94,10 @@ impl TransactionContext {
         let authenticator = self
             .authenticator
             .map(|auth| Arc::new(auth) as Arc<dyn TransactionAuthenticator>);
-        let mut tx_executor = TransactionExecutor::new(Arc::new(self.tx_inputs), authenticator);
+
+        let mut tx_executor = TransactionExecutor::new(Arc::new(self.tx_inputs), authenticator)
+            .with_debug_mode()
+            .with_tracing();
```

Comment:
Good point. I think we can revert this for now.
I'd like to improve the debugging experience and this was one requirement, but this can be done in a different way as well, either by only enabling debug and tracing if some flag is enabled or by re-executing the program with debug and tracing if the original program failed. I plan to make an issue for this.